docstore/mongodocstore/urls.go

```diff
@@ -34,27 +33,25 @@ func init() {
 // defaultDialer dials a default Mongo server based on the environment variable
 // MONGO_SERVER_URL.
 type defaultDialer struct {
-	init   sync.Once
-	opener *URLOpener
-	err    error
+	mongoServerURL string
+	opener         *URLOpener
+	err            error
 }
 
 func (o *defaultDialer) OpenCollectionURL(ctx context.Context, u *url.URL) (*docstore.Collection, error) {
-	o.init.Do(func() {
-		serverURL := os.Getenv("MONGO_SERVER_URL")
+	serverURL := os.Getenv("MONGO_SERVER_URL")
+	if serverURL != o.mongoServerURL {
```

Comment:
This doesn't work. If MONGO_SERVER_URL isn't set, then both of these will be the empty string and we'll fall through without returning an error.

How about something like `if o.opener == nil || serverURL != o.openerServerURL` ... ?

Reply:
good catch 👍🏼

i've updated this to check for an empty `MONGO_SERVER_URL` outside the if condition, so it should fail each time the env var is empty

```go
	if currentEnv == "" {
		o.err = errors.New("MONGO_SERVER_URL environment variable is not set")
		return nil, fmt.Errorf("open collection %s: %v", u, o.err)
	}

	// If MONGO_SERVER_URL has been updated, then update o.opener as well
	if currentEnv != o.mongoServerURL {
		client, err := Dial(ctx, currentEnv)
...
...
```

wdyt?